ydb/core/tablet_flat/flat_part_charge.h

```diff
@@ -235,6 +235,18 @@ namespace NTable {
                             prechargeCurrentFirstRowId = Max<TRowId>(); // no precharge
                         }
                     }
+                    if (itemsLimit && prechargeCurrentFirstRowId <= prechargeCurrentLastRowId) {
```

Comment:
Changes explanation: as soon as we decided to charge current page (prechargeCurrentFirstRowId is valid) we touch current page rows and may count them regardless of key2/endRowId